greenheart/simulation/technologies/ammonia/ammonia.py

```diff
@@ -0,0 +1,694 @@
+        non_energy_cost_in_startup_year (float): Non-energy cost in the startup year.
+        variable_cost_in_startup_year (float): Variable cost in the startup year.
+        credits_byproduct (float): Credits from byproducts.
+    """
+
+    # CapEx
+    capex_air_separation_crygenic: float
+    capex_haber_bosch: float
+    capex_boiler: float
+    capex_cooling_tower: float
+    capex_direct: float
+    capex_depreciable_nonequipment: float
+    land_cost: float
+
+    # Fixed OpEx
+    labor_cost: float
+    general_administration_cost: float
+    property_tax_insurance: float
+    maintenance_cost: float
+    total_fixed_operating_cost: float
+
+    # Feedstock and Byproduct costs
+    H2_cost_in_startup_year: float
+    energy_cost_in_startup_year: float
+    non_energy_cost_in_startup_year: float
+    variable_cost_in_startup_year: float
+    credits_byproduct: float
+
+
+@define
+class AmmoniaCostModelOutputs(AmmoniaCosts):
+    """
+    Outputs from the ammonia cost model, extending `AmmoniaCosts` with total capital
+    expenditure calculations.
+
+    Attributes:
+        capex_total (float): The total capital expenditure for the ammonia plant.
+    """
+
+    # CapEx
+    capex_total: float
+
+
+def run_ammonia_model(
+    plant_capacity_kgpy: float, plant_capacity_factor: float
+) -> float:
+    """
+    Calculates the annual ammonia production in kilograms based on the plant's
+    capacity and its capacity factor.
+
+    Args:
+        plant_capacity_kgpy (float): The plant's annual capacity in kilograms per year.
+        plant_capacity_factor (float): The capacity factor of the plant, a ratio of
+            its actual output over a period of time to its potential output if it
+            were possible for it to operate at full capacity continuously over the
+            same period.
+
+    Returns:
+        float: The calculated annual ammonia production in kilograms per year.
+    """
+    ammonia_production_kgpy = plant_capacity_kgpy * plant_capacity_factor
+
+    return ammonia_production_kgpy
+
+
+def run_ammonia_cost_model(config: AmmoniaCostModelConfig) -> AmmoniaCostModelOutputs:
+    """
+    Calculates the various costs associated with ammonia production, including
+    capital expenditures (CapEx), operating expenditures (OpEx), and credits from
+    byproducts, based on the provided configuration settings.
+
+    Args:
+        config (AmmoniaCostModelConfig): Configuration object containing all necessary
+            parameters for the cost calculation, including plant capacity, capacity
+            factor, and feedstock costs.
+
+    Returns:
+        AmmoniaCostModelOutputs: Object containing detailed breakdowns of calculated
+            costs, including total capital expenditure, operating costs, and credits
+            from byproducts.
+    """
+    feedstocks = config.feedstocks
+
+    model_year_CEPCI = 596.2  # TODO: what year
```

Comment:
Let's get this figured out soon. Either prior to merge or make an issue to make sure this does not get forgotten.